src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,24 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public static void main(String[] args) {
+        int year = 8000;
+        boolean leapYear = isLeapYear(year);
+        if (leapYear)
+            System.out.println("The year " + year + " is leap");
+        else System.out.println("The year " + year + " is not leap");
+    }
+
+    public static boolean isLeapYear (int year) {
+        boolean leapYear;
+        if (year < 1 || year > 9999)
+            leapYear = false;
+        else if (year % 4 == 0 && year % 100 != 0)
```

Comment:
This is correct, but I personally prefer doing smth like this

        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }
In your case you have 2 separate expressions where you ser identical result=true, but they could be combined into one